### PR TITLE
feat(clouddriver): Basic support for retryable kato tasks

### PR DESCRIPTION
**Pull request by robzienert**
Depends on https://github.com/spinnaker/kork/pull/383

### CODE REVIEW

**Comment by cfieber** [comment]
Thanks I Love It

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy [hunk 1]
```diff
@@ -48,10 +50,12 @@ class MonitorKatoTask implements RetryableTask {
 
   private final Clock clock
   private final Registry registry
+  @Autowired private KatoService kato
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy]: could you add this to the constructor instead

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy [hunk 2]
```diff
@@ -48,10 +50,12 @@ class MonitorKatoTask implements RetryableTask {
...
 
   @Autowired
-  public MonitorKatoTask(Registry registry) {
+  MonitorKatoTask(Registry registry) {
     this(registry, Clock.systemUTC())
+    this.kato = kato
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy]: this seems wrong without KatoService as a constructor arg

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy [hunk 3]
```diff
@@ -23,9 +23,11 @@ import com.netflix.spinnaker.orca.TaskResult
 import com.netflix.spinnaker.orca.clouddriver.KatoService
 import com.netflix.spinnaker.orca.clouddriver.model.Task
 import com.netflix.spinnaker.orca.clouddriver.model.TaskId
+import com.netflix.spinnaker.orca.pipeline.model.Execution
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import groovy.transform.CompileStatic
 import groovy.transform.TypeCheckingMode
+import io.github.resilience4j.retry.RetryConfig
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy]: I don't see this used anywhere in here, old?

File: orca-web/config/orca.yml [hunk 4]
```diff
@@ -51,6 +51,26 @@ tasks:
 logging:
   config: classpath:logback-defaults.xml
 
+resilience4j.retry:
+  instances:
+    # TODO(rz): These defaults were just copied from the resilience4j website. They still need to be tuned.
+    default:
+      maxRetryAttempts: 3
+      waitDuration: 10s
+      enableExponentialBackoff: true
```
Comment (marchello2000) [inline orca-web/config/orca.yml]: we talked in person but these defaults will blow orca's 60s task ack timeout which is going to cause issues/double executions, etc

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy [hunk 5]
```diff
@@ -149,7 +149,11 @@ class MonitorKatoTask implements RetryableTask {
       }
       katoTasks << m
       outputs["kato.tasks"] = katoTasks
+    }
 
+    if (status == ExecutionStatus.TERMINAL && katoTask.status.retryable) {
+      kato.resumeTask(katoTask.id)
+      status = ExecutionStatus.RUNNING
```
Comment (marchello2000) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy]: probably worth at least logging something here that we are "going to retry".
You are basically hiding an error from the user and it is possible the error goes away on retry or it becomes another error... and that makes it harder to debug issues. (this is the thing we talked about some standardization in orca about error messages... someday)

Comment (robzienert) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy]: You're right. I've updated the PR to show usage of the `SystemNotification` stuff I was talking about.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy (hunk 1 above)
Comment (robzienert) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy]: yeeeeeahh... I tried once and wound up reverting it. I'll try again.

**Pull request merged**